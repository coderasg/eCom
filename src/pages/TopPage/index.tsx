import React, {  useCallback, useState } from 'react';
import {   useSearchParams } from 'react-router-dom';

/**
 * Common Commponent
 */
import Header from '../../components/Header';

/**
 *  Common Hooks
 */
import useDebounce from './../../hooks/utils/useDebounce';
import objectToQueryString from './../../utils/objectToQueryString';

/**
 *  Page Specific Components
 */
import TopPageHeader from './components/TopPageHeader';
import ProductList from './components/ProductList';
import ProductCategories  from './components/ProductCategories';

/**
 * Page Specific CSS
 */
import styles from './topPage.module.scss';

function TopPage() {
  const [searchParams, setSearchParams] = useSearchParams();
  const category = searchParams.get('category') ? Number(searchParams.get('category')) : null;

  const [searchStr, setSearchStr] = useState<string>(searchParams.get('searchStr') || '' );
  const [selectedId, setSelectedId] = useState<number|null>(category);


  const onSearch = useCallback((str: string[]) => {
    setSearchStr(str[0]);

    setSearchParams(objectToQueryString({
      searchStr: str[0],
      category: selectedId
    }));
  }, []);

  const debounce = useDebounce(onSearch, 400);

  const handleOnSearchChange = useCallback(
    (value: string) => {
      debounce(value);
    },
    []
  );

  const handleTabClick = useCallback((id: number) => {
    setSelectedId(id);
    setSearchParams(objectToQueryString({
      searchStr: searchStr,
      category: id
    }));
  }, []);

  return (
    <div data-testid="TopPage-wrapper" className={styles.topPage}>
      <Header>
        <TopPageHeader handleOnSearchChange={handleOnSearchChange}  searchStr={searchStr} />
      </Header>
      <section className={styles.pageContent}>
        <ProductCategories handleTabClick={handleTabClick} selectedId={selectedId} />
        <ProductList searchStr={searchStr}  selectedId={selectedId} />
      </section>
    </div>
  );
}

export default TopPage;
