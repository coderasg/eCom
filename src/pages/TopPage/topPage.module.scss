@use "./../../variables";

.topPage {
  background-color: variables.$backgroundColor;
}
