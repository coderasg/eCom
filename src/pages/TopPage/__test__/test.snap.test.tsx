import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import TopPage from './../index';

describe('(Component)(Snapshot) TopPage', () => {
  const defaultProps = {};
    
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<BrowserRouter><TopPage {...defaultProps} /></BrowserRouter>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});