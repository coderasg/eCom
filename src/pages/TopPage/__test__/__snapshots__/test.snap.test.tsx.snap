// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component)(Snapshot) TopPage should pass snapshot test 1`] = `
<div
  data-testid="TopPage-wrapper"
>
  <div
    className="undefined undefined"
    data-testid="Header-wrapper"
  >
    <div
      data-testid="TopPageHeader-wrapper"
    >
      <section>
        <SvgIcon
          fill="#b1b0b5"
          height={25}
          width={25}
        />
        <div
          data-testid="Input-wrapper"
        >
          <input
            data-testid="searchInput"
            name="searchProduct"
            onChange={[Function]}
            placeholder="Search product"
            type="text"
            value=""
          />
        </div>
      </section>
      <section>
        <SvgIcon
          fill="#b1b0b5"
          height={25}
          width={25}
        />
        <SvgIcon
          fill="#b1b0b5"
          height={20}
          width={25}
        />
      </section>
    </div>
  </div>
  <section>
    <h1
      data-testid="ProductCategories-Error"
    >
      failed to load: ReferenceError: fetch is not defined
    </h1>
    <div
      data-testid="ProductList-Error"
    >
      <h1>
        failed to load: ReferenceError: fetch is not defined
      </h1>
    </div>
  </section>
</div>
`;
